src/ann_solo/decoy_generator.py

```diff
@@ -0,0 +1,186 @@
+from difflib import ndiff
+from typing import List, Tuple, Dict
+
+import numpy as np
+from pyteomics import parser
+from spectrum_utils import fragment_annotation, proforma
+from spectrum_utils.spectrum import MsmsSpectrum
+
+from ann_solo.config import config
+
+def _shuffle(peptide_sequence: str, excluded_residues: List[str] =['K', 'R', 'P'],
+             max_similarity: float = 0.7) -> Tuple[str, Dict[int, int]]:
+    """
+    Shuffles a peptide sequence randomly by keeping the number of tryptic
+    termini and missed internal cleavages (K,R,P). The shuffled sequence has
+    to be at least 70% dissimilar than the original sequence.
+
+    Parameters
+    ----------
+    peptide_sequence: str
+        The peptide sequence to shuffle.
+    excluded_residues: List[str]
+        The list of amino acids to maintain the tryptic property.
+
+    Returns
+    -------
+    Tuple[str, Dict[int, int]]
+        A tuple, where the first returned value is the shuffled sequence
+        and the second is the mapping indecies of the shuffle.
+    """
+    # Parse peptide
+    seq_original = parser.parse(peptide_sequence)

```

Comment:
**issue:** I think we should use the ProForma parsing functionality from spectrum_utils here instead to support standardized peptide formatting rather than Pyteomics' weird modX notation.